Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/facade/AgentFacadeImpl.java

```diff
@@ -76,27 +74,24 @@ public AgentDTO getAgentByCodeName(String codename) {
     }
 
     @Override
-    public void assignAgentToMission(AgentDTO agent, MissionDTO mission) {
-        Mission mappedMission = beanMappingService.mapTo(missionService.getEntityById(mission.getId()), Mission.class);
-        Agent mappedAgent = beanMappingService.mapTo(agent, Agent.class);
-        mappedMission.addAgent(mappedAgent);
-        agentService.save(mappedAgent);
+    public void assignAgentToMission(Long agentId, Long missionId) {
+        Mission mappedMission = beanMappingService.mapTo(missionService.getEntityById(missionId), Mission.class);
```

Comment:
Mapování je tady zbytečné. Service vrací Mision. U ostatních měněných metod platí to samé.